src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -57,6 +57,18 @@ enum ValidWiFiCredentialLength
     kWPAPSKHex = 64,
 };
 
+void AddScanResponseWithInsertionSortByRssi(chip::Platform::ScopedMemoryBuffer<ThreadScanResponse> & scanResponseArray, size_t step,
```

Comment:
It might be useful to add this as a generic utility to https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/SortUtils.h.  That would help evaluate the correctness of the algorithm as standalone code and would allow potential reuse.

Reply:
Thank you 🙂 I implemented only part of insertion sorting for performance, but as your suggestion I will improve and extract this code to `SortUtils.h` for readability and reusability.

Reply:
Thanks!